lib/asciidoctor/converter/semantic-html5.rb

```diff
@@ -49,6 +81,24 @@ def convert_inline_anchor node
     end
   end
 
+  def generate_section_numbering node
```

Comment:
While we're changing the section number output, we might want to consider offering a preview of the fix for https://github.com/asciidoctor/asciidoctor/issues/3702 (click through to the comment for the proposal). This is controlled by the converter anyway when it calls `Section#sectnum`. For parts, it should look for `partnumsuffix` first, then `sectnumsuffix`, then fall back to the current values.

I'm leaning towards `sectnumseparator` and `sectnumsuffix` (as complements to `idseparator` and `idprefix`). I had proposed dropping the `sect` prefix on the name, but someone argued that numbers in captions should be treated differently, so we're not sure about that yet.

Reply:
I've added #3702 to the [Create Modern HTML Converter project](https://github.com/asciidoctor/asciidoctor/projects/1).
I would rather keep that for a follow-up pull request.